sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/tuplesketchesAggregates.scala

```diff
@@ -0,0 +1,1895 @@
+    this(key, summary, lgNomEntries, mode, 0, 0)
+  }
+
+  /**
+   * Override inputTypes to specify key, summary (double/float), lgNomEntries (int), and mode
+   * (string) parameters.
+   */
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(
+      keyInputTypes,
+      summaryInputType,
+      IntegerType,
+      StringTypeWithCollation(supportsTrimCollation = true))
+
+  /**
+   * Override checkInputDataTypes to validate base inputs (key, summary, lgNomEntries) and mode
+   * parameter.
+   */
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val defaultCheck = checkBaseInputDataTypes()
+    if (defaultCheck.isFailure) return defaultCheck
+
+    checkModeParameter()
+  }
+
+  // Copy constructors required by ImperativeAggregate
+  override def withNewMutableAggBufferOffset(
+      newMutableAggBufferOffset: Int): TupleSketchAggDouble =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): TupleSketchAggDouble =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override protected def withNewChildrenInternal(
+      newFirst: Expression,
+      newSecond: Expression,
+      newThird: Expression,
+      newFourth: Expression): TupleSketchAggDouble =
+    copy(key = newFirst, summary = newSecond, lgNomEntries = newThird, mode = newFourth)
+
+  override def first: Expression = key
+  override def second: Expression = summary
+  override def third: Expression = lgNomEntries
+  override def fourth: Expression = mode
+
+  // Overrides for TypedImperativeAggregate
+  override def prettyName: String = "tuple_sketch_agg_double"
+
+  /** Specifies accepted summary input types (double and float). */
+  override protected def summaryInputType: AbstractDataType =
+    TypeCollection(DoubleType, FloatType)
+
+  /**
+   * Creates a DoubleSummaryFactory with the configured aggregation mode.
+   *
+   * @return
+   *   a DoubleSummaryFactory instance configured with the aggregation mode
+   */
+  override protected def createSummaryFactory(): SummaryFactory[DoubleSummary] = {
+    val mode = ThetaSketchUtils.getDoubleSummaryMode(modeInput)
+    new DoubleSummaryFactory(mode)
+  }
+
+  /**
+   * Creates DoubleSummarySetOperations for merge operations with the configured mode.
+   *
+   * @return
+   *   a DoubleSummarySetOperations instance configured with the aggregation mode
+   */
+  override protected def createSummarySetOperations(): SummarySetOperations[DoubleSummary] = {
+    val mode = ThetaSketchUtils.getDoubleSummaryMode(modeInput)
+    new DoubleSummarySetOperations(mode)
+  }
+
+  /**
+   * Converts Float inputs to Double, ensuring compatibility with DoubleSummary.
+   *
+   * @param input
+   *   the input value to normalize (Float or Double)
+   * @return
+   *   the normalized Double value
+   */
+  override protected def normalizeSummaryValue(input: Any): java.lang.Double = {
```

Comment:
Do we need this code if we can use `ExplicitCastInputTypes` instead?

Reply:
@dtenedor I used the `ImplicitCastInputTypes` trait instead of `ExpectsInputTypes`. This allows for implicit coercion with floats when using `theta_sketch_agg_double` as seen [here](https://github.com/apache/spark/blob/2bd1d48477f394bdba6ce39eeb13f22e83f731e6/sql/core/src/test/resources/sql-tests/inputs/tuplesketch.sql#L517). All summaries for `theta_sketch_agg_double` will be cast to DoubleType if that is possible, else a spark error is thrown. My only concern with this trait is that it allows some other interesting things such as passing lgNomEntries as a string `'15'` or  a double `15.0D` when it is supposed to be an integer, but it works after casting! Same for the summary types, it will cast in case of mismatch which is not always safe.